File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[C#]-Visual Studio 2010 SDK Samples/Source Code Control Provider/C#/TDD/MockTPDProvider.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VsSDK.UnitTestLibrary;
using Microsoft.VisualStudio;
using Microsoft.VisualStudio.Shell.Interop;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Samples.VisualStudio.SourceControlIntegration.SccProvider;

namespace Microsoft.Samples.VisualStudio.SourceControlIntegration.SccProvider.UnitTests
{
    class MockTrackProjectDocumentsProvider
    {
        private static GenericMockFactory tpdFactory = null;

        #region TPD Getters

        /// <summary>
        /// Return a IVsTrackProjectDocuments2 without any special implementation
		/// </summary>
		/// <returns></returns>
        internal static BaseMock GetBaseTrackProjectDocuments()
		{
            if (tpdFactory == null)
                tpdFactory = new GenericMockFactory("TPDProvider", new Type[] { typeof(IVsTrackProjectDocuments2) });

            BaseMock tpd = tpdFactory.GetInstance();
            return tpd;
		}

        /// <summary>
        /// Get an IVsTrackProjectDocuments2 that implement AdviseTrackProjectDocumentsEvents and UnadviseTrackProjectDocumentsEvents.
        /// </summary>
        /// <returns></returns>
        internal static BaseMock GetTrackProjectDocuments()
        {
            BaseMock tpd = GetBaseTrackProjectDocuments();
            string name = string.Format("{0}.{1}", typeof(IVsTrackProjectDocuments2).FullName, "AdviseTrackProjectDocumentsEvents");
            tpd.AddMethodCallback(name, new EventHandler<CallbackArgs>(AdviseTrackProjectDocumentsEventsCallBack));
            name = string.Format("{0}.{1}", typeof(IVsTrackProjectDocuments2).FullName, "UnadviseTrackProjectDocumentsEvents");
            tpd.AddMethodCallback(name, new EventHandler<CallbackArgs>(UnadviseTrackProjectDocumentsEventsCallBack));
            return tpd;
        }
        
        #endregion

		#region Callbacks

        private static void AdviseTrackProjectDocumentsEventsCallBack(object caller, CallbackArgs arguments)
        {
            uint tpdTrackProjectDocumentsCookie = 1;
            arguments.SetParameter(1, tpdTrackProjectDocumentsCookie);
            arguments.ReturnValue = VSConstants.S_OK;
        }

        private static void UnadviseTrackProjectDocumentsEventsCallBack(object caller, CallbackArgs arguments)
        {
            Assert.AreEqual((uint)1, (uint)arguments.GetParameter(0), "Incorrect cookie unregistered");
            arguments.ReturnValue = VSConstants.S_OK;
        }

    	#endregion
	}
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[VB]-Visual Studio 2010 SDK Samples/Source Code Control Provider/VB/TDD/MockTPDProvider.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.VsSDK.UnitTestLibrary
Imports Microsoft.VisualStudio
Imports Microsoft.VisualStudio.Shell.Interop
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Microsoft.Samples.VisualStudio.SourceControlIntegration.SccProvider

Namespace Microsoft.Samples.VisualStudio.SourceControlIntegration.SccProvider.UnitTests
	Friend Class MockTrackProjectDocumentsProvider
		Private Shared tpdFactory As GenericMockFactory = Nothing

		#Region "TPD Getters"

		''' <summary>
        ''' Return a IVsTrackProjectDocuments2 without any special implementation.
		''' </summary>
		''' <returns></returns>
		Friend Shared Function GetBaseTrackProjectDocuments() As BaseMock
			If tpdFactory Is Nothing Then
				tpdFactory = New GenericMockFactory("TPDProvider", New Type() { GetType(IVsTrackProjectDocuments2) })
			End If

			Dim tpd As BaseMock = tpdFactory.GetInstance()
			Return tpd
		End Function

		''' <summary>
		''' Get an IVsTrackProjectDocuments2 that implement AdviseTrackProjectDocumentsEvents and UnadviseTrackProjectDocumentsEvents.
		''' </summary>
		''' <returns></returns>
		Friend Shared Function GetTrackProjectDocuments() As BaseMock
			Dim tpd As BaseMock = GetBaseTrackProjectDocuments()
			Dim name As String = String.Format("{0}.{1}", GetType(IVsTrackProjectDocuments2).FullName, "AdviseTrackProjectDocumentsEvents")
			tpd.AddMethodCallback(name, New EventHandler(Of CallbackArgs)(AddressOf AdviseTrackProjectDocumentsEventsCallBack))
			name = String.Format("{0}.{1}", GetType(IVsTrackProjectDocuments2).FullName, "UnadviseTrackProjectDocumentsEvents")
			tpd.AddMethodCallback(name, New EventHandler(Of CallbackArgs)(AddressOf UnadviseTrackProjectDocumentsEventsCallBack))
			Return tpd
		End Function

		#End Region

		#Region "Callbacks"

		Private Shared Sub AdviseTrackProjectDocumentsEventsCallBack(ByVal caller As Object, ByVal arguments As CallbackArgs)
			Dim tpdTrackProjectDocumentsCookie As UInteger = 1
			arguments.SetParameter(1, tpdTrackProjectDocumentsCookie)
			arguments.ReturnValue = VSConstants.S_OK
		End Sub

		Private Shared Sub UnadviseTrackProjectDocumentsEventsCallBack(ByVal caller As Object, ByVal arguments As CallbackArgs)
			Assert.AreEqual(CUInt(1), CUInt(arguments.GetParameter(0)), "Incorrect cookie unregistered")
			arguments.ReturnValue = VSConstants.S_OK
		End Sub

		#End Region
	End Class
End Namespace
